script/multiNetworkExecution.sh

```
@@ -0,0 +1,1576 @@
+}
+
+function isContractAlreadyDeployed() {
+    # Check if a contract is already deployed to a network
+    local contract="$1"
+    local network="$2"
+    local environment="$3"
+
+    # Check if contract address exists in deployments file
+    local deployment_file="./deployments/${network}.json"
+    if [[ -f "$deployment_file" ]]; then
+        local contract_address=$(jq -r --arg contract "$contract" '.[$contract] // empty' "$deployment_file")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)